include/Gaffer/Loop.inl

```diff
@@ -379,8 +392,15 @@ const ValuePlug *Loop<BaseType>::sourcePlug( const ValuePlug *output, const Cont
 	}
 	else if( ancestor == outPlugInternal() )
 	{
-		const int iterations = iterationsPlug()->getValue();
-		if( iterations > 0 && enabledPlug()->getValue() )
+		int iterations;
+		bool enabled;
+		{
+			typename LoopTraits<BaseType>::IterationsScope iterationsScope( context );
```

Comment:
Seems like in theory we should also be using this scope when calling `indexVariablePlug()->getValue()` on line 375 above?